backend/estimate-service/src/main/java/org/egov/works/service/WorkflowService.java

```diff
@@ -0,0 +1,262 @@
+public class WorkflowService {
+
+    @Autowired
+    private EstimateServiceConfiguration serviceConfiguration;
+
+    @Autowired
+    private ServiceRequestRepository repository;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+
+    /*
+     * Should return the applicable BusinessService for the given request
+     *
+     */
+    public BusinessService getBusinessService(EstimateRequest estimateRequest) {
+        String tenantId = estimateRequest.getEstimate().getTenantId();
+        StringBuilder url = getSearchURLWithParams(tenantId, ESTIMATE_BUSINESSSERVICE);
+        RequestInfo requestInfo = estimateRequest.getRequestInfo();
+        Object result = repository.fetchResult(url, requestInfo);
+        BusinessServiceResponse response = null;
+        try {
+            response = mapper.convertValue(result, BusinessServiceResponse.class);
+        } catch (IllegalArgumentException e) {
+            throw new CustomException("PARSING ERROR", "Failed to parse response of workflow business service search");
+        }
+
+        if (CollectionUtils.isEmpty(response.getBusinessServices()))
+            throw new CustomException("BUSINESSSERVICE_NOT_FOUND", "The businessService " + ESTIMATE_BUSINESSSERVICE + " is not found");
+
+        return response.getBusinessServices().get(0);
+    }
+
+
+    /* Call the workflow service with the given action and update the status
+     * return the updated status of the application
+     *
+     */
+    public String updateWorkflowStatus(EstimateRequest estimateRequest) {
+        ProcessInstance processInstance = getProcessInstanceForEstimate(estimateRequest);
+        ProcessInstanceRequest workflowRequest = new ProcessInstanceRequest(estimateRequest.getRequestInfo(), Collections.singletonList(processInstance));
+        State state = callWorkFlow(workflowRequest);
+        estimateRequest.getEstimate().setEstimateStatus(state.getApplicationStatus());
+        return state.getApplicationStatus();
+    }
+
+
+    public void validateAssignee(EstimateRequest estimateRequest) {
+
+        /* Call HRMS service and validate of the assignee belongs to same department
+         * as the employee assigning it
+         *
+         */
+
+    }
+
+    /**
+     * Creates url for search based on given tenantId and businessservices
+     *
+     * @param tenantId        The tenantId for which url is generated
+     * @param businessService The businessService for which url is generated
+     * @return The search url
+     */
+
+    private StringBuilder getSearchURLWithParams(String tenantId, String businessService) {
+
+        StringBuilder url = new StringBuilder(serviceConfiguration.getWfHost());
+        url.append(serviceConfiguration.getWfBusinessServiceSearchPath());
+        url.append("?tenantId=");
+        url.append(tenantId);
+        url.append("&businessServices=");
+        url.append(businessService);
+        return url;
+    }
+
+
+    public void enrichmentForSendBackToCititzen() {
```

Comment:
There isn't any citizen involved in our system

Reply:
removed.